Sources/BSON/Document.swift

```diff
@@ -73,7 +145,12 @@ public struct Document {
      * `BSON.null`.
      */
     public subscript(key: String) -> BSON? {
-        get { fatalError("Unimplemented") }
+        get {
+            for (docKey, value) in self where docKey == key {
+                return value
+            }
+            return nil
+        }
         set { fatalError("Unimplemented") }
```

Comment:
I'm also fine with leaving the setter implementation here and below to a future PR and only supporting writing values via the initializer for the moment, but we should make a ticket for adding these

Reply:
https://jira.mongodb.org/browse/SWIFT-867
adding the todo in the next commit